data/src/main/java/com/smile/data/repository/CharacterRepositoryImpl.kt

```diff
@@ -13,6 +13,16 @@ class CharacterRepositoryImpl @Inject constructor(
 ) : CharacterRepository {
 
     override suspend fun getCharacterList(): Wrapper<List<Character>> {
+        val resp = getCharactersRemote()
+        val characterList = resp.data ?: loadCached()
```

Comment:
Nit: If your data layer is an android module, you can decide to make this reactive using livedata so that your UI queries just your local cache and the call to remote only updates the local cache. This will make the logic a bit less complex to understand. I believe you can also use Kotlin flow for this yes? (if yes, then an android module wont be required to achieve this) 🙂

Reply:
@chizoba yea, so i thought about this, my real issue is the errors, do I still return the API errors or what do I do with them? any suggestion on this?

Reply:
Can you create a git issue of this with the details so I could work on it? @chizoba